src/tls.c

```diff
@@ -176,17 +175,51 @@ enftun_tls_handshake(struct enftun_tls* tls)
 }
 
 int
-enftun_tls_connect(struct enftun_tls* tls, const char** hosts, const char* port)
+enftun_tls_connect(struct enftun_tls* tls, const char** hosts, const char* port, int fwmark)
 {
     int rc;
 
-    rc = tls->sock->ops.connect_any(tls->sock, hosts, port);
-    if (rc < 0)
-        goto out;
+    enftun_log_info("%s Enter tls->sock=%p tls->sock_type=%d\n",
+        __func__, tls->sock, tls->sock_type);
+
+    /* If XTT hasn't dec */
+    if (!tls->sock) {
```

Comment:
This is overly complex branching.  There are three different calls to `connect_any`, when there are only two possible choices (native and SCM).  (Especially since we're no longer going to try to remember which type XTT used.)

Reply:
The logic in `xtt.c` is much simpler:  Try the native socket. If that fails, and SCM is enabled, try SCM.

Reply:
I was worried different methods causing errors, if it's only covering an edge optimization I 100% agree on removing this.